homeassistant/exceptions.py

```diff
@@ -64,6 +64,12 @@ def __str__(self) -> str:
             return self._message
 
         if not self.generate_message:
+            # Initialize self._message to prevent a recursive loop
+            self._message = super(HomeAssistantError).__str__()  # type: ignore[misc]
```

Comment:
```suggestion
            if type(self) is not HomeAssistantError:
                self._message = super(HomeAssistantError).__str__()  # type: ignore[misc]
```

This can only happen if its subclassed?

Reply:
Yes

Reply:
But may be the first call should just be a static assignment, like the name of the (sub) class.

Reply:
It is a bit unexpected to make a function call and than throw it away

Reply:
It is not thrown away if the second call is a recursive one

Reply:
Changed the first call to `self._message = str(self.__class__)`. The tests show the result of the first call is not thrown away.